src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -24,19 +24,21 @@
  */
 package org.spongepowered.api.plugin;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.regex.Pattern;
 
 /**
  * An annotation used to describe and mark a Sponge plugin.
  */
-@Target(TYPE)
-@Retention(RUNTIME)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
 public @interface Plugin {
 
+    Pattern ID_PATTERN = Pattern.compile("[a-z0-9-_]+");
```

Comment:
People might change the ID as well, we just have to trust that they won't. I am personally not in favor of identifying plugins by a String because it's too unreliable. Using classes for dependencies also just looks cleaner to my eye in general. A class is pretty much the same as a "group" and "id" if you think about it also.